app/src/main/java/com/mybus/view/HistoryViewHolder.java

```diff
@@ -26,7 +27,12 @@ public HistoryViewHolder(View itemView, HistoryItemSelectedListener itemSelected
     @Override
     public void onClick(View view) {
         if (mItemSelectedListener != null) {
-            mItemSelectedListener.onHistoryItemSelected(historyAddress.getText().toString());
+            mItemSelectedListener.onHistoryItemSelected(mRecentLocation);
```

Comment:
In my opinion and from I saw on the examples the ViewHolder shouldn't have an private reference to the object which represent. The ViewHolder should return the position and the adpater return the object indeed.